packages/config/src/class-config.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * The base Jetpack configuration class file.
+ *
+ * @package automattic/jetpack-config
+ */
+
+namespace Automattic\Jetpack;
+
+use Automattic\Jetpack\Sync\Actions as Sync_Actions;
+use Automattic\Jetpack\Sync\Main as Sync_Main;
+use Automattic\Jetpack\Plugin\Tracking as Plugin_Tracking;
+use Automattic\Jetpack\Terms_Of_Service;
+
+/**
+ * The configuration class.
+ */
+class Config {
+
+	/**
+	 * The Jetpack class instance.
+	 *
+	 * @var Jetpack
+	 */
+	protected $jetpack;
+
+	/**
+	 * The initial setting values.
+	 *
+	 * @var Array
+	 */
+	protected $config = array(
+		'sync'                => false,
+		'sync_woocommerce'    => false,
+		'sync_wp_super_cache' => false,
+		'tracking'            => false,
+		'tos'                 => false,
+	);
+
+	/**
+	 * Creates the configuration class instance, initalized with the Jetpack object.
+	 *
+	 * @param \Jetpack $jetpack the main object to initalize everything with.
+	 */
+	public function __construct( \Jetpack $jetpack ) {
+		$this->jetpack = $jetpack;
+
+		add_action( 'plugins_loaded', array( $this, 'on_plugins_loaded_early' ), 5 );
+		add_action( 'plugins_loaded', array( $this, 'on_plugins_loaded' ) );
+		add_action( 'plugins_loaded', array( $this, 'on_plugins_loaded_late' ), 90 );
+	}
+
+	/**
+	 * Runs early on plugins_loaded hook execution.
+	 *
+	 * @action plugins_loaded
+	 */
+	public function on_plugins_loaded_early() {
+
+		/**
+		 * Filters the settings for the Jetpack Config package, allowing package consumers to set
+		 * initialization options.
+		 *
+		 * @since 8.1.0
+		 *
+		 * @param Array An array of options that configure Jetpack package initialization details.
+		 */
+		$this->config = wp_parse_args(
+			apply_filters( 'jetpack_config', $this->config ),
+			$this->config
+		);
+
+		if ( $this->config['sync'] ) {
+			Sync_Main::init();
```

Comment:
Since `Main::init()` will now be run during `plugins_loaded`, I wonder if we can remove 'Automattic\Jetpack\Sync\Main'  from the [autoloader's ignore list](https://github.com/Automattic/jetpack/blob/master/packages/autoloader/src/autoload.php#L77)? 

I did a few quick tests with Sync\Main removed from the ignore list, and I didn't see any Notices.

Reply:
Done, thanks!